### PR TITLE
removing new lines while showing url path field 

### LINKED ISSUE #1526: [Bug] API URL field allows multiline input causing big twitch in UI
97785414-dbce3480-1bca-11eb-891f-b0628d7cea16

### Steps to reproduce the behaviour:
1. Go to API pane
2. Enter **multiline** input in URL field
3. Toggle between *focus* and *blur* on the field

### LINKED ISSUE #3621: [Bug] API Pane URL does not contract after it loses focus
Screenshot 2021-03-18 at 8 45 38 PM

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description
if url path contains '\n', field is not able to do line wrapping properly

Fixes #1526
Fixes #3621

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 1]
```diff
@@ -183,38 +183,54 @@ class CodeEditor extends Component<Props, State> {
...
-      }
-      if ((!!inputValue || inputValue === "") && inputValue !== editorValue) {
-        this.editor.setValue(inputValue);
+      const inputValue = this.getInputValueUpdated(textEditorValue);
+      if (!!inputValue || inputValue === "") {
+        if (this.props.size === "COMPACT") {
+          this.editor.setValue(this.removedNewLinesFromUrl(inputValue));
+        } else {
+          this.editor.setValue(inputValue);
+        }
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:226]: abstract this in a common function

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 2]
```diff
@@ -183,38 +183,54 @@ class CodeEditor extends Component<Props, State> {
...
       }
       this.updateMarkings();
 
       // if (currentMode !== this.props.mode) {
       //   this.editor.setOption("mode", this.props?.mode);
       // }
     } else {
-      // Update the dynamic bindings for autocomplete
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:217]: Revert deleting this comment

**Review by hetunandu — CHANGES_REQUESTED** [review]
Needs refactoring

**Comment by hetunandu** [comment]
> No changes to code coverage between the base branch and the head branch

Consider adding a jest test for this functionality. Mock the underlying code mirror or input props and assert if the right functions are called

File: app/client/src/components/editorComponents/CodeEditor/index.tsx (hunk 1 above)
Comment (ApekshaBhosale) [reply app/client/src/components/editorComponents/CodeEditor/index.tsx]: for this I have changed a condition a little bit so abstracting won't make much sense.

**Comment by ApekshaBhosale** [comment]
> > No changes to code coverage between the base branch and the head branch
> 
> Consider adding a jest test for this functionality. Mock the underlying code mirror or input props and assert if the right functions are called

@hetunandu which functionality you are talking about? the whole thing or just particular function I created?

**Comment by hetunandu** [comment]
> > > No changes to code coverage between the base branch and the head branch
> > 
> > 
> > Consider adding a jest test for this functionality. Mock the underlying code mirror or input props and assert if the right functions are called
> 
> @hetunandu which functionality you are talking about? the whole thing or just particular function I created?

Anything that can get us started with testing this component

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 3]
```diff
@@ -195,18 +193,15 @@ class CodeEditor extends Component<Props, State> {
...
-        typeof inputValue === "string"
-      ) {
-        inputValue += "";
-      }
-      if ((!!inputValue || inputValue === "") && inputValue !== editorValue) {
-        this.editor.setValue(inputValue);
+      const inputValue = getInputValue(this.props.input.value);
+      if (!!inputValue || inputValue === "") {
+        if (this.props.size === EditorSize.COMPACT) {
+          console.log(inputValue);
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:200]: remove console.log
```suggestion

```

**Review by hetunandu — DISMISSED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**